### PR TITLE
add libmysqlclient/8.0.17

### PR DESCRIPTION
**Pull request by ericLemanissier**
this is largely based on mysql-connector-c

**libmysqlclient/8.0.17**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by lasote** [comment]
Not working even adjusting the SDKROOT. It might be a different issue.

**Comment by ericLemanissier** [comment]
I have no idea what is happening here

**Comment by lasote** [comment]
So, we have some news, A `\n` at the end of the value of the `SDKROOT` variable was messing with us. Now that we have it correctly working we still have 2 issues:

1. Messages like: `ld: warning: object file (/tmp/c3ipr/pr_172_15_0_0/.conan/data/zlib/1.2.11/_/_/package/4f892a0d0107b794d837128fbfc6812497252e9f/lib/libz.a(compress.o)) was built for newer OSX version (10.14) than being linked (10.13)` We have to regenerate binaries for the other packages
2. The build still fails, probably this is not related to the SDK issues:
```
dyld: Library not loaded: libmysqlclient.21.dylib

  Referenced from: /private/tmp/c3ipr/pr_172_15_0_0/.conan/data/libmysqlclient/8.0.17/_/_/build/486f3dfa64d71f7a484786080c4854a07b26f5e7/libmysql/../runtime_output_directory/libmysql_api_test

  Reason: image not found
```

**Comment by SSE4** [comment]
the second one is definitely not a SDK issue - it's recipe problem (with `rpath`, `DYLD_LIBRARY_PATH` or `install_name_tool`)

**Comment by ericLemanissier** [comment]
CI does not seem to produce any results. Is it coming from the recipe ?

File: recipes/libmysqlclient/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        os.rename("CMakeLists.txt", sources_cmake)
+        if self.settings.os == "Macos":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "libmysql", "CMakeLists.txt"), "COMMAND $<TARGET_FILE:libmysql_api_test>", "COMMAND DYLD_LIBRARY_PATH=%s $<TARGET_FILE:libmysql_api_test>" % os.path.join(self.build_folder, "library_output_directory"))
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self.options.shared:
+            raise ConanInvalidConfiguration("libmysqlclient cannot be built as static library")
+        if self.settings.compiler == "Visual Studio"
```
Comment (danimtb) [inline recipes/libmysqlclient/all/conanfile.py]: you have a missing `:` here:
```suggestion
        if self.settings.compiler == "Visual Studio":
```

Comment (ericLemanissier) [reply recipes/libmysqlclient/all/conanfile.py]: That's embarrassing. thanks!

File: recipes/libmysqlclient/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+project(cmake_wrapper)
+message(WARNING "Conan libMySQLclient Wrapped CMake")
+
+include(../conanbuildinfo.cmake)
+conan_basic_setup(NO_OUTPUT_DIRS)
+include("CMakeListsOriginal.txt")
+
+# Turn off -Werror in GCC debug builds.
+if(CMAKE_COMPILER_IS_GNUCC AND CMAKE_COMPILER_IS_GNUCXX)
```
[the remark this answers is not in the document]
Comment (ericLemanissier) [reply recipes/libmysqlclient/all/CMakeLists.txt]: this is a remnant of mysql-connector-c package

**Pull request merged**